### PR TITLE
[BANDWIDTH_THROTTLING] Update Chargeable interface to align it with QuotaChargeCallback and updated QuotaManager behavior.

### PR DESCRIPTION
**Pull request by ankagrawal**
This PR updates the Chargeable interface to align it with QuotaChargeCallback and updated QuotaManager behavior.
The QuotaAwareOperationController that uses this Chargeable interface is also updated to use new interfaces.

### CODE REVIEW

**Review by justinlin-linkedin — APPROVED** [review]
the rest looks good to me

**Comment by justinlin-linkedin** [comment]
And just a thought: we are only using QuotaChargeCallback because we need RestRequest object and we only expecting 1: Account 2 Container 3 Method from RestRequest. It seems like we can. easily get these three pieces of information from router. We have blob id passed to every operations so we can get account id and container id thus we can get Account and Container and in each operation we know the method, GetBlobOperation means it's QuoteMethod.Read, PutOperation means it's QuotaMethod.Write. In this way, we can probably get rid of `QuotaChargeCallback` so the code becomes more clear.

**Review by tbingol — APPROVED** [review]
Please provide a PR description

File: ambry-router/src/main/java/com/github/ambry/router/GetBlobOperation.java [hunk 1]
```diff
@@ -673,7 +670,7 @@ void reset() {
       decryptJobMetricsTracker = new CryptoJobMetricsTracker(routerMetrics.decryptJobMetrics);
       correlationIdToGetRequestInfo.clear();
       state = ChunkState.Free;
-      isCharged = false;
+      operationQuotaCharger = new OperationQuotaCharger(quotaChargeCallback, "GetBlobOperation", routerMetrics);
```
[the remark this answers is not in the document]
Comment (ankagrawal) [reply ambry-router/src/main/java/com/github/ambry/router/GetBlobOperation.java]: done.

**Pull request merged**